exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -114,119 +151,43 @@ func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config)
-
-	// 3. Each histogram bucket is converted to a cumulative counter called
-	// <basename>_bucket and will include a dimension called upper_bound that
-	// specifies the maximum value in that bucket. This metric specifies the
-	// number of events with a value that is less than or equal to the upper
-	// bound.
-	bucketMetricName := metricName + separator + bucketSuffix + separator
-	explicitBuckets := distributionValue.BucketOptions.GetExplicit()
-	if explicitBuckets == nil {
-		return values, fmt.Errorf(
-			"unknown bucket options type for metric %q",
-			bucketMetricName)
-	}
-	bounds := explicitBuckets.Bounds
-	splunkBounds := make([]*string, len(bounds)+1)
-	for i := 0; i < len(bounds); i++ {
-		dimValue := float64ToDimValue(bounds[i])
-		splunkBounds[i] = &dimValue
-	}
-	splunkBounds[len(splunkBounds)-1] = &infinityBoundSFxDimValue
-
-	for i, bucket := range distributionValue.Buckets {
-		values[fmt.Sprintf("%s%s", bucketMetricName, *splunkBounds[i])] = bucket.Count
+func copy(fields map[string]interface{}) map[string]interface{} {
+	newFields := make(map[string]interface{}, len(fields))
+	for k, v := range fields {
+		newFields[k] = v
 	}
-
-	return values, nil
-}
-
-func float64ToDimValue(f float64) string {
-	// Parameters below are the same used by Prometheus
-	// see https://github.com/prometheus/common/blob/b5fe7d854c42dc7842e48d1ca58f60feae09d77b/expfmt/text_create.go#L450
-	// SignalFx agent uses a different pattern
-	// https://github.com/signalfx/signalfx-agent/blob/5779a3de0c9861fa07316fd11b3c4ff38c0d78f0/internal/monitors/prometheusexporter/conversion.go#L77
-	// The important issue here is consistency with the exporter, opting for the
-	// more common one used by Prometheus.
-	str := strconv.FormatFloat(f, 'g', -1, 64)
-	return str
+	return newFields
 }
 
-func mapSummaryValue(metric *metricspb.Metric, summaryValue *metricspb.SummaryValue) (map[string]interface{}, error) {
-
-	// Translating summary values per symmetrical recommendations to Prometheus:
-	// https://docs.signalfx.com/en/latest/integrations/agent/monitors/prometheus-exporter.html#overview
-
-	// 1. The total count gets converted to a cumulative counter called
-	// <basename>_count.
-	// 2. The total sum gets converted to a cumulative counter called <basename>
-	values := make(map[string]interface{})
-	metricName := fmt.Sprintf("%s:%s", splunkMetricValue, metric.GetMetricDescriptor().Name)
-	values[metricName+separator+countSuffix] = summaryValue.Count.Value
-	values[metricName] = summaryValue.Sum.Value
-
-	// 3. Each quantile value is converted to a gauge called <basename>_quantile
-	// and will include a dimension called quantile that specifies the quantile.
-	percentiles := summaryValue.GetSnapshot().GetPercentileValues()
-	if percentiles == nil {
-		return values, fmt.Errorf(
-			"unknown percentiles values for summary metric %q",
-			metricName)
+func timestampToEpochMilliseconds(ts pdata.TimestampUnixNano) *float64 {
```

Comment:
Nit: the name is a bit misleading. It returns the timestamp in seconds with millisecond precision.

Reply:
Fixed